aten/src/ATen/native/cpu/CatKernel.cpp

```diff
@@ -45,11 +45,17 @@ void cat_serial_kernel_impl(Tensor& result, TensorList tensors, int64_t dim) {
           result_ptr[k] = input_ptr[k];
         }
       } else {
-        vec::map(
-            [](Vec x) { return x; },
-            result_ptr,
-            input_ptr,
-            local_inner);
+        int64_t d = 0;
```

Comment:
Seems to be, unrelated change, can you put it as separate PR

Reply:
Indeed this is unrelated, but since i changed functional specification for bfloat16. The behavior of `map<BFloat16>` changed:
* (before): `map<BFloat16>` would require the vec type to be `Vectorized<BFloat16>`
* (after): `map<BFloat16>` would require the vec type to be `Vectorized<float>`

Using `Vectorized<float>` is aiming at reducing the unnecessary dtype change between fp32 and bf16. AKA. only input and output data will do the dtype change and only fp32 is used in the computation for better perf and less rounding error.

So if is not changed, there would be compilation error.

However, after the change of `map<BFloat16>` behavior, it is no longer suitable for `cat` which is only a simple memory copy (no computation).

if we stick to use `map<BFloat16>`, it is going to be:
```
1. load as bf16 vec (length = 16)
2. convert into 2 * fp32 vec (length = 2 * 8)
3. convert back to bf16 vec (length = 16)
4. store to dst address
```

This is not our intention. A simple load/store will be enough here. So for the simple memcpy case it is no longer suitable to use `map<>` for BFloat16. I haven't notice other similar circumstances (using `map<>` for memcpy on BFloat16), will modify if met any.